### PR TITLE
fix: RND-77: Create idiomatic method names for sdk

### PR DESCRIPTION
**Pull request by niklub**
The following API<>SDK mapping is supported:
```csv
API,SDK
GET /api/projects,projects.list()
POST /api/projects,projects.create()
GET /api/projects/{id},projects.get()
DELETE /api/projects/{id},projects.delete()
PATCH /api/projects/{id},projects.update()
GET /api/projects/{id}/next,projects.next_task()
GET /api/tasks,tasks.list()
POST /api/tasks,tasks.create()
POST /api/projects/{id}/import,tasks.create_many()
GET /api/projects/{id}/import,tasks.create_many_status()
GET /api/tasks/{id},tasks.get()
DELETE /api/tasks/{id},tasks.delete()
PATCH /api/tasks/{id},tasks.update()
GET /api/annotations,annotations.list()
POST /api/annotations,annotations.create()
GET /api/annotations/{id},annotations.get()
DELETE /api/annotations/{id},annotations.delete()
PATCH /api/annotations/{id},annotations.update()
GET /api/predictions,predictions.list()
POST /api/predictions,predictions.create()
GET /api/predictions/{id},predictions.get()
DELETE /api/predictions/{id},predictions.delete()
PATCH /api/predictions/{id},predictions.update()
GET /api/projects/{id}/file-uploads,projects.files.list()
DELETE /api/projects/{id}/file-uploads,projects.files.delete_many()
GET /api/data/upload/{filename},projects.files.download()
GET /api/projects/{id}/file-uploads/{id},projects.files.get()
DELETE /api/projects/{id}/file-uploads/{id},projects.files.delete()
PATCH /api/projects/{id}/file-uploads/{id},projects.files.update()
GET /api/ml,ml.list()
POST /api/ml,ml.create()
GET /api/ml/{id},ml.get()
DELETE /api/ml/{id},ml.delete()
PATCH /api/ml/{id},ml.update()
POST /api/ml/{id}/train,ml.train()
POST /api/ml/{id}/interactive-annotating,ml.predict_interactive()
GET /api/ml/{id}/versions,ml.list_model_versions()
GET /api/storages,import_storage.list()
GET /api/storages/types,import_storage.list_types()
GET /api/storages/export,export_storage.list()
GET /api/storages/export/types,export_storage.list_types()
GET /api/storages/[TYPE],import_storage.[TYPE].list()
POST /api/storages/[TYPE],import_storage.[TYPE].create()
GET /api/storages/[TYPE]/{id},import_storage.[TYPE].get()
DELETE /api/storages/[TYPE]/{id},import_storage.[TYPE].delete()
PATCH /api/storages/[TYPE]/{id},import_storage.[TYPE].update()
POST /api/storages/[TYPE]/sync,import_storage.[TYPE].sync()
POST /api/storages/[TYPE]/validate,import_storage.[TYPE].validate()
GET /api/storages/[TYPE]/export,export_storage.[TYPE].list()
POST /api/storages/[TYPE]/export,export_storage.[TYPE].create()
GET /api/storages/[TYPE]/export/{id},export_storage.[TYPE].get()
DELETE /api/storages/[TYPE]/export{id},export_storage.[TYPE].delete()
PATCH /api/storages/[TYPE]/export/{id},export_storage.[TYPE].update()
POST /api/storages/[TYPE]/export/sync,export_storage.[TYPE].sync()
POST /api/storages/[TYPE]/export/validate,export_storage.[TYPE].validate()
GET /api/storages/[TYPE]/export,export_storage.[TYPE].list()
GET /api/users,users.list()
POST /api/users,users.create()
GET /api/users/{id},users.get()
DELETE /api/users/{id},users.delete()
PATCH /api/users/{id},users.update()
POST /api/current-user/reset-token,users.reset_token()
GET /api/current-user/token,users.get_token()
GET /api/current-user/whoami,users.whoami()
GET /api/labels,labels.list()
POST /api/labels,labels.create()
GET /api/labels/{id},labels.get()
DELETE /api/labels/{id},labels.delete()
PATCH /api/labels/{id},labels.update()
GET /api/label_links,projects.labels.list()
POST /api/label_links,projects.labels.create()
POST /api/labels/bulk,projects.labels.create_many()
GET /api/label_links/{id},projects.labels.get()
DELETE /api/label_links/{id},projects.labels.delete()
PATCH /api/label_links/{id},projects.labels.update()
GET /api/organizations,organizations.list()
GET /api/organizations/{id}/memberships,organizations.members.list()
DELETE /api/organizations/{id}/memberships,organizations.members.delete()
GET /api/organizations/{id},organizations.get()
PATCH /api/organizations/{id},organizations.update()
GET /api/invite,organizations.get_invite()
GET /api/invite/reset-token,organizations.reset_invite()
```

### CODE REVIEW

File: label_studio/labels_manager/api.py [hunk 1]
```diff
@@ -27,6 +27,8 @@
     name='create',
     decorator=swagger_auto_schema(
         tags=['Labels'],
+        x_fern_sdk_group_name='labels',
```
Comment (jombooth) [inline label_studio/labels_manager/api.py:30]: should these be `['projects', 'labels']`? I do see that the operation description mentions projects.

File: label_studio/ml/api.py [hunk 2]
```diff
@@ -20,11 +20,33 @@
...
+    },
+    required=[],
+)
+
 
 @method_decorator(
     name='post',
     decorator=swagger_auto_schema(
         tags=['Machine Learning'],
+        x_fern_sdk_group_name='ml',
```
Comment (jombooth) [inline label_studio/ml/api.py:47]: should this also be in the `projects` group? same question as https://github.com/HumanSignal/label-studio/pull/5925/files#r1628528060 really

File: label_studio/organizations/api.py [hunk 3]
```diff
@@ -240,6 +256,8 @@ def get(self, request, *args, **kwargs):
     name='post',
     decorator=swagger_auto_schema(
         tags=['Invites'],
+        x_fern_sdk_group_name='organizations',
+        x_fern_sdk_method_name='reset_invite',
```
Comment (jombooth) [inline label_studio/organizations/api.py]: Should this be `reset_token`?

File: label_studio/projects/api.py [hunk 4]
```diff
@@ -92,6 +92,91 @@
...
+        'color': openapi.Schema(
+            title='color',
+            description='Project color in HEX format',
+            type=openapi.TYPE_STRING,
+            default='#FFFFFF',
+        ),
+        'control_weights': openapi.Schema(
+            title='control_weights',
+            description='Dict of weights for each control tag in metric calculation. Each control tag (e.g. label or choice) will '
+            "have it's own key in control weight dict with weight for each label and overall weight."
```
Comment (jombooth) [inline label_studio/projects/api.py]:
```suggestion
            'have its own key in control weight dict with weight for each label and overall weight. '
```

File: label_studio/projects/api.py [hunk 5]
```diff
@@ -92,6 +92,91 @@
...
+            'For example, if bounding box annotation with control tag named my_bbox should be included with 0.33 weight in agreement calculation, '
+            'and the first label Car should be twice more important than Airplaine, then you have to need the specify: '
+            "{'my_bbox': {'type': 'RectangleLabels', 'labels': {'Car': 1.0, 'Airplaine': 0.5}, 'overall': 0.33}",
```
Comment (jombooth) [inline label_studio/projects/api.py]:
```suggestion
            'For example, if a bounding box annotation with a control tag named my_bbox should be included with 0.33 weight in agreement calculation, '
            'and the first label Car should be twice as important as Airplane, then you need to specify: '
            "{'my_bbox': {'type': 'RectangleLabels', 'labels': {'Car': 1.0, 'Airplane': 0.5}, 'overall': 0.33}",
```

File: label_studio/projects/api.py [hunk 6]
```diff
@@ -574,6 +677,7 @@ class ProjectReimportAPI(generics.RetrieveAPIView):
     name='get',
     decorator=swagger_auto_schema(
         tags=['Projects'],
+        x_fern_audiences=['internal'],  # TODO: deprecate this endpooint in favor of tasks:tasks-list
```
Comment (jombooth) [inline label_studio/projects/api.py]:
```suggestion
        x_fern_audiences=['internal'],  # TODO: deprecate this endpoint in favor of tasks:tasks-list
```

File: label_studio/ml/api.py [hunk 7]
```diff
@@ -216,7 +251,10 @@ def post(self, request, *args, **kwargs):
     name='post',
     decorator=swagger_auto_schema(
         tags=['Machine Learning'],
-        operation_summary='Predict',
+        x_fern_sdk_group_name='ml',
+        x_fern_sdk_method_name='test_predict',
+        x_fern_audiences=['internal'],
```
Comment (jombooth) [inline label_studio/ml/api.py:256]: Should this really be internal?

File: label_studio/io_storages/s3/api.py [hunk 8]
```diff
@@ -38,7 +80,13 @@
 @method_decorator(
     name='post',
     decorator=swagger_auto_schema(
-        tags=['Storage:S3'], operation_summary='Create new storage', operation_description='Get new S3 import storage'
+        tags=['Storage:S3'],
+        x_fern_sdk_group_name=['import_storage', 's3'],
+        x_fern_sdk_method_name='create',
+        x_fern_audiences=['public'],
+        operation_summary='Create new S3 storage',
+        operation_description='Get new S3 import storage',
```
Comment (jombooth) [inline label_studio/io_storages/s3/api.py]: These seem a little contradictory, though it looks like that problem existed before this PR

File: label_studio/io_storages/all_api.py [hunk 9]
```diff
@@ -76,6 +80,8 @@ def get(self, request, **kwargs):
     name='get',
     decorator=swagger_auto_schema(
         tags=['Storage'],
+        x_fern_sdk_group_name='import_storage',
+        x_fern_sdk_method_name='list',
```
Comment (jombooth) [inline label_studio/io_storages/all_api.py:113]: should `projects` group be added here?

File: label_studio/io_storages/all_api.py [hunk 10]
```diff
@@ -108,6 +114,8 @@ def list(self, request, *args, **kwargs):
     name='get',
     decorator=swagger_auto_schema(
         tags=['Storage'],
+        x_fern_sdk_group_name='export_storage',
+        x_fern_sdk_method_name='list',
```
Comment (jombooth) [inline label_studio/io_storages/all_api.py:148]: and here? https://github.com/HumanSignal/label-studio/pull/5925/files?diff=unified&w=0#r1628559192

File: label_studio/data_manager/api.py [hunk 11]
```diff
@@ -349,6 +430,9 @@ def get(self, request):
     name='post',
     decorator=swagger_auto_schema(
         tags=['Data Manager'],
+        x_fern_sdk_group_name='actions',
```
Comment (jombooth) [inline label_studio/data_manager/api.py:433]: should this be `['actions', 'views']`

File: label_studio/data_import/api.py [hunk 12]
```diff
@@ -588,8 +620,9 @@ def delete(self, request, *args, **kwargs):
     name='delete',
     decorator=swagger_auto_schema(
         tags=['Import'],
-        x_fern_sdk_group_name='files',
+        x_fern_sdk_group_name=['projects', 'files'],
```
Comment (jombooth) [inline label_studio/data_import/api.py]: I think these actually should not be in the projects group; the file upload API doesn't seem to have anything project specific in it

File: label_studio/data_import/api.py [hunk 13]
```diff
@@ -619,7 +652,14 @@ class UploadedFileResponse(generics.RetrieveAPIView):
 
     @override_report_only_csp
     @csp(SANDBOX=[])
-    @swagger_auto_schema(auto_schema=None)
+    @swagger_auto_schema(
+        tags=['Import'],
+        x_fern_sdk_group_name=['projects', 'files'],
```
Comment (jombooth) [inline label_studio/data_import/api.py]: I don't think this should be in `projects` either

**Review by jombooth — COMMENTED** [review]
A few small findings, overall looks great!

File: label_studio/labels_manager/api.py (hunk 1 above)
Comment (niklub) [reply label_studio/labels_manager/api.py:30]: `Labels` are created globally and then linked to the projects via `LabelLinkAPI`.
On the other hand, we never productionalized the entire workflow with labels and user-defined custom categories in Taxonomy (especially right now when it is under `legacy` parameter). I propose to hide this endpoint in SDK until we readdress ontology management etc.

File: label_studio/ml/api.py (hunk 7 above)
Comment (niklub) [reply label_studio/ml/api.py:256]: I'd hide everything that is partially implemented, and release as it is ready and clearly defined. In this endpoint, the most straightforward flow to submit prediction payload is not implemented yet.

File: label_studio/io_storages/all_api.py (hunk 9 above)
Comment (niklub) [reply label_studio/io_storages/all_api.py:113]: it doesn't seem to work at all actually, only `/types` endpoints are working. Shall we remove these ones?
the `/storages/types` endpoint returns the list of globally available types of storage, unrelated to project. I'll leave them under `client.import_storage.list_types()`

File: label_studio/io_storages/all_api.py (hunk 10 above)
Comment (niklub) [reply label_studio/io_storages/all_api.py:148]: same as https://github.com/HumanSignal/label-studio/pull/5925#discussion_r1629201461

File: label_studio/data_import/api.py (hunk 12 above)
Comment (niklub) [reply label_studio/data_import/api.py]: Agree, in general, it is better to follow the endpoint design with less hierarchy, to ensure we can rearrange objects in more global names, for example, cloud storage or model connections in projects vs. workspaces etc.

**Review by jombooth — APPROVED** [review]

File: label_studio/ml/api.py (hunk 2 above)
Comment (niklub) [reply label_studio/ml/api.py:47]: we keep method names with minimal hierarchy to make it clear and avoid potential refactoring if one day we decide that ML models should also live one level above the projects

**Pull request merged**